### DISCUSSION TITLE
GitHub Copilot Chat doesn't work if NordVPN is connected.

### DISCUSSION THREAD

**Discussion by ZhuStar** [community/community#107463] [Copilot Conversations] [answered]
### Select Topic Area

Question

### Body

I need to pause NordVPN for Copilot Chat to work. As a brute force fix, I tried Split Tunneling **everything** under Visual Studio in the Task Manager, turned off _Web Protection_, _File Protection_, and _Vulnerability Detection_ but that still didn't fix the problem, so there's probably another process not running under Visual Studio which I need to add to the Split Tunneling.

These are the processes added to Split Tunneling:
- Visual Studio (devenv.exe)
- Microsoft.ServiceHub.Controller
- Node.js JavaScript Runtime
- PerfWatson2.exe
- ServiceHub.Host.AnyCPU
- ServiceHub.Host.dotnet.x64
- ServiceHub.Host.netfx.x86
- ServiceHub.IdentityHost.exe
- ServiceHub.IndexingService.exe
- ServiceHub.IntellicodeModelService.exe
- ServiceHub.RoslynCodeAnalysisService.exe
- ServiceHub.TestWindowStoreHost.exe
- ServiceHub.ThreadWaitDialog.exe
- ServiceHub.VSDetouredHost.exe

My Copilot works normally with the VPN, but it shows these messages (using the /log command-line argument to devenv.exe):

```
03:47:08.970 [Verbose] stderr:  [ApplicationInsights:Sender []
03:47:08.970 [Verbose] stderr:  [  'Ingestion endpoint could not be reached. This batch of telemetry items has been lost. Use Disk Retry Caching to enable resending of failed telemetry. Error:',]
03:47:08.970 [Verbose] stderr:  [  []
03:47:08.970 [Verbose] stderr:  [    "[object Error]{ stack: 'Error: getaddrinfo ENOTFOUND copilot-telemetry.githubusercontent.com\n" +]
03:47:08.970 [Verbose] stderr:  [      "    at GetAddrInfoReqWrap.onlookup [as oncomplete] (node:dns:84:26)', message: 'getaddrinfo ENOTFOUND copilot-telemetry.githubusercontent.com', name: 'Error'"]
03:47:08.970 [Verbose] stderr:  [  ]]
03:47:08.970 [Verbose] stderr:  []]
```

Copilot Chat displays this message when I ask it a question with the VPN connected but everything else turned off (_Web Protection_, _File Protection_, and _Vulnerability Detection_) note that for me to ask anything to Copilot Chat with the VPN connected, I have to disconnect the VPN while opening Copilot Chat window, otherwise it just loads forever with the message [Conversations Information] [CopilotClient] Need to get an auth token:

```
[Conversations Information] [CopilotClient] Need to get an auth token.
[CopilotSessionProvider Information] Cancelled sending message (ConversationId:d9e817b8-8d5c-4957-a0c5-331748962646, CorrelationId:d5d414ac-c50f-4cc8-8877-f7f8ca6476bd, MessageId: 7ce4065c-65db-43d9-bcb9-e5d2cd35fa8e)
```

There's a situation which I'm having trouble reproducing in which the error includes this:
```
Microsoft.VisualStudio.Conversations.Service.ServiceTarget.AskInConversationAsync(CopilotSessionId sessionId, CopilotRequest request, ActiveInteractions activeInteractions, CancellationToken cancellation) in D:\a\_work\1\s\src\Conversations.Service.Implementation\ServiceTarget.cs:line 133
```

### System Details:
**Windows:**
Edition	Windows 11 Pro
Version	23H2
OS build	22631.3085
Experience	Windows Feature Experience Pack 1000.22684.1000.0

**Visual Studio:**
Microsoft Visual Studio Enterprise 2022 (64-bit) - Current
Version 17.8.6

**NordVPN:**
7.18.5.0

Edit: Currently, I have no extensions enabled other than Copilot and Copilot Chat, to avoid any conflict.

**Comment by ayim** [comment, marked as answer]
This networking related issue looks like the sort of thing where we may need to collect more logs.
Please file a bug using the Report a Problem feature in the top right of VS.

![image](https://github.com/community/community/assets/10951869/bb36a914-5746-416f-8140-5aaa57c35b8a)

**Comment by ThomasLeopold2022** [comment]
Any news on this? I have the same issue.
When I send a question to Copilot Chat, I get this in Visual Studio's Output area:
[Conversations Information] [CopilotCompletions] Error while processing the event: The requested address is not valid in its context. (api.githubcopilot.com:443)
Automatically retrying, please check if your internet connection is working ...

However, when I ping api.githubcopilot.com, even from within Visual Studio via Developer PowerShell, I get replies.

I should mention that signing in to Copilot Chat seems to work now.

**Reply by ayim** [reply]
To confirm, is the expected behaviour that Copilot Chat works with NordVPN and the behaviour you're encountering is that Copilot Chat is NOT working with NordVPN enabled?

**Reply by ThomasLeopold2022** [reply]
Yes, you are correct,
Copilot Chat does not work when I'm connected via NordVPN. I have tried all three protocols (NordLynx, OpenVPN (UDP), and OpenVPN (TCP)), different server types and locations and a lot more. Even exempting Visual Studio 2022 from the VPN connection using NordVPN's split tunneling feature doesn't help. I keep getting:

[Conversations Information] [CopilotCompletions] Error while processing the event: The requested address is not valid in its context. (api.githubcopilot.com:443)
Automatically retrying, please check if your internet connection is working ...

when I send a question to Copilot Chat. It shows "Thinking ..." in the chat window itself, and the error message (multiple times) in VS's output area.
Only if I completely deactivate the VPN connection, I can communicate with Copilot Chat.

**Reply by ayim** [reply]
Gotcha. Please use this mechanism so that we can get the ball rolling and analyze additional telemetry for your use case.

![image](https://github.com/community/community/assets/10951869/667cd2b1-d4aa-4808-a500-859ec71ace8f)

**Reply by ThomasLeopold2022** [reply]
Here's the thread: https://developercommunity.visualstudio.com/t/GitHub-Copilot-Chat-runs-into-error-when/10610977

**Comment by thiago-marquezini** [comment]
The backend API is probably blocking the vpn connection.

**Reply by ThomasLeopold2022** [reply]
That's what needs to be found out. If it's done on purpose, I'm going to cancel my paid subscription, because I can't use it.

**Comment by SiqiLu** [comment]
I've encountered this issue as well and spent a lot of time trying to resolve it. I found that if you disable NordVPN's Split Tunneling and turn off the DNS-related options in Threat Protection, Copilot works normally with NordVPN proxy. I'm not entirely sure about the reason behind this.

Just for reference, I'm using Visual Studio 2022 with GitHub Copilot. I hope this helps!

**Reply by SiqiLu** [reply]
Using my real IP address for anything makes me feel unsafe.

**Reply by ThomasLeopold2022** [reply]
I independently found out just the same: A simple VPN connection to NordVPN works just fine with Copilot, as long as split tunneling is fully disabled.
Interestingly, It looks like connecting to a cloud Team Foundation Server (from Visual Studio 2022) and the Windows synchronization of the PC clock (e. g. via Settings) also fail, when split tunneling is enabled.
